Centaurus.Domain/Quanta/Processors/Payments/PaymentRequestProcessorBase.cs

```diff
@@ -58,7 +58,7 @@ public override void Validate(MessageEnvelope envelope)
             if (payment.Amount <= 0)
                 throw new InvalidOperationException("Amount should be greater than 0");
 
-            if (!Global.Constellation.Assets.Any(a => a.Id == payment.Asset))
+            if (payment.Asset != 0 && !Global.Constellation.Assets.Any(a => a.Id == payment.Asset))
```

Comment:
Iterating through all assets may be the most expensive check in the validation routine. Let's optimize it a bit. Store all available asset ids as a `HashSet<int>` in the global settings. Then you can do a search with O(1) complexity: `global.AssetIds.Contains(payment.Asset)`. The difference is negligible when you have only a few assets, but may be substantial once total assets count exceeds a dozen.
Do not forget to recreate the set every time the constellation settings are changed.